app/code/Magento/CatalogInventory/Model/StockIndex.php

```diff
@@ -217,17 +227,19 @@ protected function getWebsitesWithDefaultStores($websiteId = null)
      *
      * @param int $productId
      * @param int $websiteId
-     * @return $this
+     * @return void
      */
     protected function processParents($productId, $websiteId)
     {
-        $parentIds = [];
+        $parentIds = [[]];
         foreach ($this->getProductTypeInstances() as $typeInstance) {
             /* @var $typeInstance AbstractType */
-            $parentIds = array_merge($parentIds, $typeInstance->getParentIdsByChild($productId));
+            $parentIds[] = $typeInstance->getParentIdsByChild($productId);
         }
 
-        if (!$parentIds) {
+        $parentIds = array_merge(...$parentIds);
+
+        if (empty($parentIds)) {
             return $this;
```

Comment:
I think it should be like this to be valid
```suggestion
            return;
```

Reply:
Nice catch!